### PR TITLE
[fix] correct skips in design rule checks (```test_common.py```) caused by fragile whitelist_to_blacklist

### PR DESCRIPTION
**Pull request by icfaust**
## Description

fix errors in private CI and Azure Pipelines caused by differences in ```sys.path```.  This PR will enable those tests properly in those CI systems.  It deals with possible issues with user paths and symlinks, as well as erroneous inclusions of subdirectories (if they occur).

No benching necessary as it is purely testing code.

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/tests/test_common.py [hunk 1]
```diff
@@ -49,6 +49,54 @@
     "IncrementalEmpiricalCovariance-score-call_validate_data",  #  must call clone of itself
     "SVC(probability=True)-fit-call_validate_data",  #  SVC fit can use sklearn estimator
     "NuSVC(probability=True)-fit-call_validate_data",  #  NuSVC fit can use sklearn estimator
+    "LogisticRegression-score-n_jobs_check",  # uses daal4py for cpu in sklearnex
```
Comment (samir-nasibli) [inline sklearnex/tests/test_common.py:52]: We only have one list for all design violations vs all design rules?
Should be renamed for the specific rule I think and better to have for the specific rule specific maybe list of design violation.

Comment (icfaust) [reply sklearnex/tests/test_common.py:52]: The design rule is in the string, I'm not sure I understand

Comment (samir-nasibli) [reply sklearnex/tests/test_common.py:52]: I mean we have only one skip list for the all design rules?

Comment (samir-nasibli) [reply sklearnex/tests/test_common.py:52]: I mean test should be parameterized like this
`def test_estimator(estimator, method, design_pattern, design_violation_list, estimator_trace):
`
Probably some design patterns should have some part of common design violation list, that should be defined.
But if we expect that all rules will not have the specific violations except some common list then make sense to leave as is

Comment (samir-nasibli) [reply sklearnex/tests/test_common.py:52]: But probably the comment for the further update :)

Comment (samir-nasibli) [reply sklearnex/tests/test_common.py:52]: Thank you @icfaust !
We have discussed it somehow, assuming we make the comment resolved.
We have one skip list for each test cases, where the design_rule is specified. Im good with this one.

Just for the further optimization we can add the string comment into pytest,xfail. to see this in the logs not only in code base. But this refactoring not the scope of the current PR.

**Review by samir-nasibli — APPROVED** [review]

File: sklearnex/tests/test_common.py (hunk 1 above)
Comment (icfaust) [reply sklearnex/tests/test_common.py]: Done, it will now give an informative string on xfail.

**Review by david-cortes-intel — APPROVED** [review]
Cannot comment on whether the list covers everything that it should, but can confirm that it solves test failures for me, at least in my local environment.

**Comment by icfaust** [comment]
I'm going to merge this and will be responsible for private CI failures.
